base/abstractarraymath.jl

```diff
@@ -487,11 +525,21 @@ julia> collect(eachslice(M, dims=2))
  [3, 6, 9]
 ```
 """
-@inline function eachslice(A::AbstractArray; dims)
-    length(dims) == 1 || throw(ArgumentError("only single dimensions are supported"))
-    dim = first(dims)
-    dim <= ndims(A) || throw(DimensionMismatch("A doesn't have $dim dimensions"))
-    inds_before = ntuple(d->(:), dim-1)
-    inds_after = ntuple(d->(:), ndims(A)-dim)
-    return (view(A, inds_before..., i, inds_after...) for i in axes(A, dim))
+@inline eachslice(A;dims) = _eachslice(A,dims)
+
+@inline _eachslice(A,dim::Integer) = _eachslice(A,(dim,))
+@inline function _eachslice(A::AbstractArray{N}, dims::NTuple{M,Integer}) where {N,M}
+    for (i,dim) in enumerate(dims)
```

Comment:
Worth noting that you could do this with recursion instead to get type stability under constant propagation, c.f. the current version of JuliennedArrays.

Reply:
Good point.